### PR TITLE
discovery: Detect instrumentation with dd-trace-go v2

### PR DESCRIPTION
**Pull request by vitkyrka**
### What does this PR do?

    v2 of dd-trace-go changed the import path from
    gopkg.in/DataDog/dd-trace-go.v1/ to github.com/DataDog/dd-trace-go/v2/,
    so the prefix match that we were using fails to match that.  Instead,
    look for the package name in the symbol name ignoring the domain.

### Motivation

https://datadoghq.atlassian.net/browse/USMON-1230

### Describe how to test/QA your changes

Automated tests have been updated.

### Additional Notes

With this change, the GetAnySymbol..() function is about 2% slower on /usr/bin/docker (which doesn't have instrumentation).

```
BenchmarkGoPrefix-20                 313          18895292 ns/op
BenchmarkGoInfix-20                  309          19304874 ns/op
```

### CODE REVIEW

File: pkg/network/go/bininspect/symbol_filter.go [hunk 1]
```diff
@@ -69,34 +69,36 @@ func (f stringSetSymbolFilter) findMissing(symbolByName map[string]safeelf.Symbo
 	return missingSymbols
 }
 
-// prefixSymbolFilter is a symbol filter which gets any one symbol which has the
-// specified prefix.
-type prefixSymbolFilter struct {
-	prefix    string
+// infixSymbolFilter is a symbol filter which gets any one symbol which has the
+// specified infix.
```
Comment (guyarb) [inline pkg/network/go/bininspect/symbol_filter.go:73]: > which gets any one symbol

I think the documentation should be rephrased

File: pkg/network/go/bininspect/symbol_filter.go [hunk 2]
```diff
@@ -69,34 +69,36 @@ func (f stringSetSymbolFilter) findMissing(symbolByName map[string]safeelf.Symbo
...
-func (f prefixSymbolFilter) getNumWanted() int {
+func (f infixSymbolFilter) getNumWanted() int {
 	return 1
 }
 
-func (f prefixSymbolFilter) want(symbol string) bool {
-	return strings.HasPrefix(symbol, f.prefix)
+func (f infixSymbolFilter) want(symbol string) bool {
+	return strings.Contains(symbol, f.infix)
 }
```
[the remark this answers is not in the document]
Comment (vitkyrka) [reply pkg/network/go/bininspect/symbol_filter.go:98]: The `GetAnySymbol..()` function is about 2% slower on `/usr/bin/docker`.

```
BenchmarkGoPrefix-20                 313          18895292 ns/op
BenchmarkGoInfix-20                  309          19304874 ns/op
```

**Review by guyarb — APPROVED** [review]

File: pkg/network/go/bininspect/symbol_filter.go (hunk 1 above)
Comment (vitkyrka) [reply pkg/network/go/bininspect/symbol_filter.go]: Done.

File: pkg/network/go/bininspect/symbol_filter.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (vitkyrka) [reply pkg/network/go/bininspect/symbol_filter.go:98]: Done.

**Pull request merged**